homeassistant/components/mqtt/__init__.py

```diff
@@ -375,12 +375,14 @@ async def async_publish_service(call: ServiceCall) -> None:
         payload_template = call.data.get(ATTR_PAYLOAD_TEMPLATE)
         qos: int = call.data[ATTR_QOS]
         retain: bool = call.data[ATTR_RETAIN]
+        rendered_topic: template.Template | None = None
         if msg_topic_template is not None:
             try:
-                rendered_topic: Any = template.Template(
-                    msg_topic_template, hass
-                ).async_render(parse_result=False)
-                msg_topic = valid_publish_topic(rendered_topic)
+                rendered_topic = cv.template(msg_topic_template)
```

Comment:
I use `cv.template` to prenet a call to an untyped context.

Reply:
I think we should attempt to add typing to `template.Template`

Reply:
Reverted with # type: ignore[no-untyped-call]